### PR TITLE
implementing that authenticated users can create, get and delete blogs

### PR DESCRIPTION
**Pull request by isilacar**

### CODE REVIEW

File: src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java [hunk 1]
```diff
@@ -86,9 +96,41 @@ public List<BlogDto> getBlogsByTagName(String tagName) {
...
     public List<SimplifiedBlogResponse> getSimplifiedBlogs() {
         List<Blog> blogs = blogRepository.findAll();
-        return blogs.stream().map(blog -> new SimplifiedBlogResponse(blog.getTitle(),blog.getText())).toList();
+        return blogs.stream().map(blog -> new SimplifiedBlogResponse(blog.getTitle(), blog.getText())).toList();
+    }
+
+    @Override
+    public UserResponse deleteUserBlog(Long blogId, Long userId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new ResourceNotFound("User Not Found"));
+        Blog blog = user.getBlogList().stream()
```
Comment (ditsinski) [inline src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java:105]: there should be simplier solution to this. Think of a method to fetch blog by userName & blog id, and simply delete it if exist. Isnt this going to be much easier ?

File: src/main/java/com/scalefocus/blog_api/controller/BlogController.java [hunk 2]
```diff
@@ -20,48 +23,60 @@ public class BlogController {
...
-    public ResponseEntity<List<SimplifiedBlogResponse>> getSimplifiedBlogs(){
-        return new ResponseEntity<>(blogService.getSimplifiedBlogs(),HttpStatus.OK);
+    public ResponseEntity<List<SimplifiedBlogResponse>> getSimplifiedBlogs() {
+        return new ResponseEntity<>(blogService.getSimplifiedBlogs(), HttpStatus.OK);
     }
 
+    //authenticated users can remove own blogs
+    @DeleteMapping("/{blogId}/users/{userId}")
+    public ResponseEntity<UserResponse> deleteUserBlog(@PathVariable Long blogId, @PathVariable Long userId) {
+        return new ResponseEntity<>(blogService.deleteUserBlog(blogId, userId), HttpStatus.OK);
```
Comment (ditsinski) [inline src/main/java/com/scalefocus/blog_api/controller/BlogController.java:78]: there is no need for customer response with user Id and blog Id here. Status OK is enough

File: src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java [hunk 3]
```diff
@@ -86,9 +96,41 @@ public List<BlogDto> getBlogsByTagName(String tagName) {
...
+        userResponse.setBlogList(user.getBlogList().stream()
+                .map(b -> new BlogDto(b.getId(), b.getTitle(), b.getText(), tagDtoSet))
+                .collect(Collectors.toList()));
+        return userResponse;
+
     }
+    @Override
+    public List<UserBlogResponse> getUsersBlogs() {
 
+        return userRepository.findAll().stream().map(user -> {
```
Comment (ditsinski) [inline src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java:124]: I believe this piece of code will fetch all blogs for all users ? We want the functionality to fetch blog by username

**Review by ditsinski — CHANGES_REQUESTED** [review]

File: src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java (hunk 1 above)
Comment (isilacar) [reply src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java:105]: I thought that user has to get all the blogs of all the users. I changed it to get blogs with username.

Comment (isilacar) [reply src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java:105]: I got the point now. I changed it to get blog by username and delete it.

File: src/main/java/com/scalefocus/blog_api/controller/BlogController.java (hunk 2 above)
Comment (isilacar) [reply src/main/java/com/scalefocus/blog_api/controller/BlogController.java:78]: Done.

File: src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java (hunk 3 above)
Comment (isilacar) [reply src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java:124]: I thought that user can get all the blogs of all the users. I changed it to get blogs of specific user.

**Review by ditsinski — APPROVED** [review]

**Pull request merged**